general edits

tgbugs opened the pull request:
Ongoing PR to simplify discussion about edits to the prose of the site. Do not merge yet!

tgbugs committed:
index.org edit for readability

tgbugs committed:
install.org edits

tgbugs committed:
quickstart edits and regular nomenclature

As it turns out the preamble doesn't have a name in the org syntax
document, so I picked one. This can change if there is a preferred
nomeclature.

I changed the use of the term heading to headline since that is the
term that is used everywhere in the documentation. Might as well get
new users initiated with the correct terminology.

tgbugs committed:
features.org first pass edits for syntax and code sections

tgbugs committed:
quickstart add sections for comments and macros at the end

These are a bit more complex than might fit the quickstart guide, but
they are two things that I expect that users would like to know.
Therefore I put them at the end so that the complexity doesn't
distract from the basics, and hopefully whets some appetites.

tgbugs committed:
features edits for sections, tasks, and tables

I switched the nomenclature to TODO item for tasks since that is more
in line with nomenclature for the headlines that define the task. Not
entirely satisfied with the flow of the section on tasks.

tecosaur commented on install.org:
```diff
@@ -4,46 +4,53 @@
-[[https://github.com/syl20bnr/spacemacs/blob/master/layers/%2Bemacs/org/packages.el][Spacemacs]]. It's worth noting that /Doom/ installs a version of Org close to the
-development ~HEAD~.
+The Emacs starter kits /Doom/ and /Spacemacs/ come with a curated set of packages and configurations that may be
+more familiar to users coming from other editors (e.g. Vim). They include Org along with integrations between
```
Minor niggle (I'm a Doom user :stuck_out_tongue:) I feel that Doom/Spacemacs have utility seperate from 'familiarity layers' with other editors.

tgbugs replied:
If this comes through as them being only familiarity layers that is definitely not intended. I think I need to add something to the first sentence that captures the fact that they are definitely more than just layers since the mention of curated packages is not sufficient.

tecosaur replied:
The way I read it the start was fine "come with a curated set of packages and configurations" but then I found "that may be
more familiar to users coming from other editors" potentially confusing - as (maybe I'm looking too hard?) I saw an interpretation that in Doom/Spacemacs the whole point of the packages/configurations is to make them more familiar to users from other editors.
Shouldn't take more than a small tweak to iron this out though I don't think.

tgbugs replied:
Made a bigger change around this due to including Scimax, but also addressed this issue.

tecosaur commented on install.org:
```diff
@@ -4,46 +4,53 @@
 
-Popular configs like [[https://github.com/purcell/emacs.d/blob/master/lisp/init-org.el][Purcell's emacs.d]] will also often include org-mode
-customisations, and Org-related packages.
+Popular configs like [[https://github.com/purcell/emacs.d/blob/master/lisp/init-org.el][Purcell's emacs.d]]
```
Hmm. Does removing the "after" give the impression that Doom/Spacemacs just install packages and don't configure Org much? Maybe I'm being paranoid.

tgbugs replied:
Maybe prefixing it with something like "For an approach that isn't quite as invasive, popular configs ..."

Also reminder to include Scimax on the list of starter kits.

tecosaur replied:
Ah yes, Scimax also deserves a mention. One more niggle, I'm not a fan of the negative connotations of "invasive" (even though the denotation definately applies) - perhaps a term like "expansive" could be a improvement?

tgbugs replied:
Agree, not happy with invasive, will try to come up with a better way to make the distinction between starter kits and configs.

tgbugs replied:
I think the new version here is clearer. Can you double check it to make sure no negative language crept in?

tecosaur replied:
This reads far better to me, nice job :+1:

tecosaur commented on quickstart.org:
```diff
@@ -4,46 +4,73 @@
+* Markup
+To mark up text in Org, simply surround it with one or more marker characters.
+*bold* /italic/ and _underline_ are fairly intuitive, and the ability to use
++strikethrough+ is a plus.  You can +_/*combine*/_+ the basic markup in any
```
I'm not sure that striking `+_/*combine*/_+` is a good idea, though I do appreciate that it demonstrates 4 layers of markup instead of 3.

tgbugs replied:
I wondered about this as well. It seemed like it could confuse but wanted to check. Thus with check, will remove.

tecosaur commented on quickstart.org:
```diff
@@ -55,57 +82,97 @@ To buy:
 #+END_SRC
 
-Lists can also contain checkboxes: ~[ ]~, ~[-]~, ~[X]~
+Lists can contain checkboxes ~[ ]~, ~[-]~, ~[X]~.
```
Do you think it's worth mentioning the meaning associated with these, or is it completely obvious? e.g.
```
~[ ]~ (not started), ~[-]~ in progress, ~[X]~ completed
```
Just a thought

tgbugs replied:
Yes will add since I had no idea what the `[-]` form meant.

tecosaur commented on quickstart.org:
```diff
@@ -55,57 +82,97 @@ To buy:
-add a description? Just add a ~[description]~ after the target, i.e.
-~[[target][desc]]~.
+To create a link put the target between two square brackets, like so: ~[[target]]~.
+To add a description add it in square brackets after the target ~[[target][desc]]~.
```
If we could not have two "add"s so close to each other I think that could improve the flow of this sentence.
How about replacing the second one with "insert"?

tgbugs replied:
"To add a description include it in square ..."?

tecosaur replied:
Just the first alternative that came to mind for me, "To add a description insert it in square brackets after the target...".
Quite minor, but I'm not sure that "To **add** a ... **add** ..." flows that well.

Hmmm, maybe better to just reword the sentance and remove the 'it' as well, something like:
"You can insert a description after the target in square brackets".

tecosaur commented on quickstart.org:
```diff
@@ -55,57 +82,97 @@ To buy:
-heading which matches ~target~.
+Org supports a wide variety of types of link targets, and you can also
+[[https://orgmode.org/manual/Adding-Hyperlink-Types.html][add your own]].
+Link types are indicated with a ~type:~ prefix, e.g. ~[type:target]~.
```
Minor niggle: This should be `i.e.` not `e.g.`.

tgbugs replied:
`i.e.,` iirc but yes, since there isn't any other way to do it so it is not an example.

tecosaur commented on quickstart.org:
```diff
@@ -55,57 +82,97 @@ To buy:
 #+END_SRC
 
-If you want syntax highlighting and execution try
+However, example blocks do not give you the awesome power of [[https://orgmode.org/worg/org-contrib/babel/][Org babel]].
```
Personal taste, but how about "phenomenal" over "awesome"?

tgbugs replied:
I don't think that phenomenal works in this context. Unfortunately it does evoke the "Like 10,000 hotdogs sir." definition of awesome. Maybe just "do not give you the power of"?

tecosaur replied:
That seems like the best approach to me :slightly_smiling_face: I just didn't quite feel that "awesome" fit that well.

tecosaur commented on features.org:
```diff
@@ -54,24 +58,44 @@ More on [[file:manual/Markup-for-Rich-Contents.html][the markup syntax]] (manual
-technique a handy and natural way to deal with code.
+Org makes it trivial to extract source code from an Org file and
+generate documentation from the same file. This means that it is
+possible to have a single source of truth for implementation and
+documentation.
 
-Source code can be trivially extracted from an org file, and rich documentation
-produced from the very same file.
+# Org mode makes it possible to write executable documentation.
+# It also makes it possible to sync source code into back into an Org file. 
+# The whole Ferret compiler is implemented in this way as a single Org file.
 
+# Source code is not trapped in documentation 
```
Starts of fantastically succinct and impactful, but I feel that the second half isn't as valuable. I may be splitting hairs with this though - just something I wanted to mention.

tgbugs replied:
Agree. This needs another pass, I'm not satisfied with the 2nd section, and I have left a bunch of stuff commented since it didn't fit the flow.

tgbugs replied:
Still not satisfied, but at least the 3 sections are clearly delineated.

tecosaur commented on index.org:
```diff
@@ -103,51 +103,58 @@ novice]] and [[http://howardism.org/Technical/Emacs/literate-devops.html][powerf
-Once you've used Org for a bit, you'll know it's great --- it's actually
-so great that there have been [[https://orgmode.org/worg/org-papers.html][(scientific) papers]] written about the value of Org
-for [[https://en.wikipedia.org/wiki/Reproducibility#Reproducible_research][reproducible research]].
+Nearly every Org user has a story to tell about how Org enables and empowers them ---
```
Hmm. Is this an opportunity to turn "story" "to" and "tell" into links to such stories? Just an idea.

tgbugs replied:
Yes. This was one of the places where I really wanted to be able to link to a gallery or other user stories.

tecosaur replied:
Cool. I haven't got any links on hand, perhaps this is something to come back to later?

tgbugs replied:
I think we are just going to leave the gallery out for now.

tecosaur commented on index.org:
```diff
@@ -103,51 +103,58 @@ novice]] and [[http://howardism.org/Technical/Emacs/literate-devops.html][powerf
-so great that there have been [[https://orgmode.org/worg/org-papers.html][(scientific) papers]] written about the value of Org
-for [[https://en.wikipedia.org/wiki/Reproducibility#Reproducible_research][reproducible research]].
+Nearly every Org user has a story to tell about how Org enables and empowers them ---
+some have found it so useful that they have written [[https://orgmode.org/worg/org-papers.html][(scientific) papers]] about
```
As above re: "story to tell", perhaps we could do the same with "they have written" - or maybe I'm overeager.

tgbugs replied:
Since this is already in Worg I think we can leave it as is. As a stop gap for the gallery, maybe it could be implemented as part of Worg so that the links stay inside the orgmode site?

tecosaur replied:
This does seem less significant, perhaps we should just leave this be at least for now.

tecosaur reviewed (commented):
Hi Tom. This is brilliant! Thank you so much for taking this on. Reading through the work you've done so far has prompted a few thoughts which I've added for your perusal.

Looking great!

tgbugs committed:
edits to install features and quickstart to address comments

Still not satisfied with the two additional sections about org babel.

Removed the paragraph on org babel from quickstart since it is now the
leading paragraph in the features section on org babel.

Added Scimax to the list of starter kits.

tgbugs committed:
clean up some of the old community.org file

I think this is probably not used?

tgbugs committed:
contribute.org edits

tgbugs commented:
@tecosaur do you have the build/deployment this documented anywhere so I can do tight loop for things like layout and spacing?

tecosaur commented:
Yep, that's in the colophon page: https://orgmode.tecosaur.com/colophon.html#development

tecosaur commented:
It's basically just
```elisp
          (let ((org-html-style-default "")
                (org-html-scripts ""))
            (org-html-export-to-html))
```
Then `rsync`ing the generated files to my web server.

tecosaur reviewed (commented):
I think we could actually just delete this file (community) - it seems redundant at this point, and the twitter bot / Gsoc info would be better on Worg IMO.

tgbugs commented:
Two thoughts.

1. Can you convert the animations from gif to webm (and/or record new versions in webm if you redo some of them)? The gifs are giant compared to the webm files and will cost users and hosting bandwidth and $$.
2. It might be fun to show off publishing the website using the org publishing functionality. I have a set up that runs as a commit hook for my thesis that I could probably adapt for this.

tecosaur commented:
1. I recorded using emacs-gif-screencast, I think this could probably support webm with a little modification. I'm not sure how much improvement to the file sizes there would be given the gifs are currently quite tiny: 21, 78, 81, 150, 155, 945 KiB, or 1.4 MiB in total.
2. WDYM? I used to have links to the org sources of each page but Bastien vetoed that.

tgbugs commented:
1. Hrm, I'm able to take exports.gif from 945K to 524K using the following without loosing perceptual quality. It might be possible to tweak it a bit more https://trac.ffmpeg.org/wiki/Encode/VP9.
```
ffmpeg -i exports.gif -c:v libvpx-vp9 -b:v 0 -crf 40 -pass 1 -an -f null /dev/null && \
ffmpeg -i exports.gif -c:v libvpx-vp9 -b:v 0 -crf 40 -pass 2 exports.webm
```
2. Nothing visible to the user, just using `org-publish-project-alist` and something like the following (definitely not working and probably incorrect).
```elisp
'(("orgsite"
         :components ("html" "resources"))
        ("html"
         :base-directory "/path/to/orgwebsite"
         :base-extension "org"
         :include "*.org"
         :publishing-directory "/ssh:server:/var/www/orgsite/"
         :publishing-function org-latex-publish-to-html)
        ("resources"
         :base-directory "./resources/"
         :publishing-directory "/ssh:server:/var/www/orgsite/resources/"
         ))
```

tecosaur commented:
1. Hmm. Tried some of the other formats and reproduced the size savings. Recapturing would probably be better though. I'm a bit burnt out on this for now, but if you're interested I can happily share the details required to reproduce these.
2. I've never used org-publish before but that looks interesting.

tgbugs commented:
1. No worries. You've put a lot of time and effort in here! I have all the current gifs and can take a shot at tweaking them, I also imagine that the animations could be written functions or macros and played (if they weren't already).
2. Yep. Your existing workflow is fine for now (possibly better because iirc org-publish doesn't use rsync, rather some crazy hand rolled things with tramp), just seemed like an opportunity to consider.

tecosaur commented:
The current state looks good to me. @tgbugs would you say this is good to merge, or is there more you want to do?

tgbugs commented:
I'm doing another pass on features.

tgbugs committed:
features.org more edits

I've updated the headlines to be more aliterative, if they are too
much we can revert, but I think that it is ok to be a bit gradiose,
given that we are barely scratching the surface of the functionality
that Org has to offer.

tgbugs committed:
update simple-syntax.svg to be text based for easier modification

since it needed a reword once, it might need one again

tgbugs committed:
features.org reordered the sections for better flow

The changes make the order closer to the current site and are chunked
into broader categories for editing and structure, code, export,
planning, extensibility, and mobile.

tgbugs committed:
features.org more edits

tecosaur commented on features.org:
```diff
@@ -216,25 +245,25 @@ More on [[file:manual/Publishing.html][publishing]] (manual)
   :CUSTOM_ID: extensibiliy
   :HTML_CONTAINER_CLASS: side-fig
   :END:
+# Extremely extensible
```
I think I might preffer this (Extremely extensible) to "Emacs Extensible" - given that I want this page to make sense to people who don't use Emacs

tecosaur commented on features.org:
```diff
@@ -243,68 +272,72 @@ on remote machines with [[https://www.emacswiki.org/emacs/TrampMode][Tramp]], an
+Plan your day, your way, with ease.
 
-Org provides easy keyboard shortcuts to /clock in/ and /clock out/ of tasks.
+With /agenda views/ --- reports generated from a set of tasks --- you
```
Minor tweak, I'd change ` --- reports generated from a set of tasks --- ` to ` ---reports generated from a set of tasks--- `.

tgbugs replied:
Ah, I see that I am not up on my em dash punctuation rules.

tecosaur commented on features.org:
```diff
@@ -315,19 +348,20 @@ Tutorial on [[https://orgmode.org/worg/org-tutorials/org-custom-agenda-commands.
 
-You can capture from everywhere, both within Emacs and from other
-applications like your web browser, PDF viewer, etc.
+You can capture from anywhere -- from an Emacs buffer or from other
```
Should be `---` not `--`.

tecosaur commented:
I deleted a comment? What? If I accidently deleted something you said ... would you mind re-posting it?

I'm just going to re-do my coments on a the latest version as part of a single review. Sorry about any notification spam.

tgbugs commented:
No worries. I was commenting about the em dashes. I rarely use them, so was not completely aware of their punctuation rules. It seems that there are two options, either spaces on both sides, or no spaces according to https://www.thepunctuationguide.com/em-dash.html#spaces. I'm hunting down all the en dashes and replacing them. I think we should go with AP style for spaces so that the org word tokenizer doesn't get confused. Thoughts?

tecosaur commented on features.org:
```diff
@@ -89,113 +57,134 @@ List of [[https://orgmode.org/worg/org-contrib/babel/languages.html][supported l
-seamlessly expand individual sections to read and edit using the =tab= key.
+The fundamental structure of an Org document is a tree, with nested,
+collapsible sections forming its backbone.  You can fold and unfold
+whole trees with single keystroke -- =tab=.
```
Should be `---` not `--`.

tecosaur commented:
Ah, thanks. My understanding for best practices with dashes is:
```
- for hypen-ation
-- for ranges: 4--6, east--west
--- for (1) indicate a break in ideas (like commas, semicolons):
          things have changed a lot --- mainly for the better
        (2) to seperate a non-essential part of the sentance (like parenthesis):
          one can ---and arguably should--- do something
```

tecosaur commented on features.org:
```diff
@@ -89,113 +57,134 @@ List of [[https://orgmode.org/worg/org-contrib/babel/languages.html][supported l
+Org includes a powerful table editor.
 
-All done with just plain text markup.
+=tab= moves between columns. =return= moves between rows.
```
Maybe turn this into one sentence? "=tab= moves between columns, and =return= moves between rows."
Just a thought.

Oh, and should we maybe use =TAB= and =RET= ?

tgbugs replied:
I think that including `, and` in this context makes it harder to understand what we are trying to convey, because it forces the sentence to have two subjects, rows and columns, rather than one subject per sentence which is easier to parse.

The Emacs manual uses lowercase https://www.gnu.org/software/emacs/manual/html_node/elisp/Function-Keys.html.

tecosaur replied:
Seems reasonable

tecosaur commented on features.org:
```diff
@@ -89,113 +57,134 @@ List of [[https://orgmode.org/worg/org-contrib/babel/languages.html][supported l
-cookies, tags, properties, etc.
+Org tables aren't just static views of data.  They can be used as a
+full [[https://orgmode.org/manual/The-Spreadsheet.html][spreadsheet]]
+system, with formulas, references, and more -- all with the simplicity
```
Should be `---` not `--`.

tecosaur commented on features.org:
```diff
@@ -89,113 +57,134 @@ List of [[https://orgmode.org/worg/org-contrib/babel/languages.html][supported l
+[[https://en.wikipedia.org/wiki/Literate_programming][literate programming]]
+environment, allowing you to weave the inputs and outputs of multiple
+languages and runtime environments across multiple machines into a
+single unified system, empowering you to achieve new levels of DRYness
```
Maybe have "DRYness" link to https://en.wikipedia.org/wiki/Don%27t_repeat_yourself or similar?

tgbugs replied:
Good idea.

tecosaur commented on features.org:
```diff
@@ -89,113 +57,134 @@ List of [[https://orgmode.org/worg/org-contrib/babel/languages.html][supported l
+from the same file. While many languages have support for generating
+developer documentation from code comments, with Org your developer
+guide and user manual can also be the source code for implementation
+and testing -- a single source of truth that won't drift out of sync.
```
Should be `---` not `--`.

tecosaur commented on features.org:
```diff
@@ -89,113 +57,134 @@ List of [[https://orgmode.org/worg/org-contrib/babel/languages.html][supported l
 ** Description
 
-Org provides great capability as an authoring and publishing tool.
+Org is an authoring and publishing tool -- it can function as
```
Should be `---` not `--`.

tecosaur commented on features.org:
```diff
@@ -205,96 +194,101 @@ More on [[file:manual/Publishing.html][publishing]] (manual)
+Plan your day, your way, with ease.
 
-Org provides easy keyboard shortcuts to /clock in/ and /clock out/ of tasks.
+With /agenda views/ --- reports generated from your /TODO items/ ---
```
Consider using the ` ---XYZ--- ` style (personal taste).

tgbugs replied:
I stuck with the AP style for consistency across all em dashes.

tecosaur commented on features.org:
```diff
@@ -305,19 +299,44 @@ Tutorial on [[https://orgmode.org/worg/org-tutorials/org-custom-agenda-commands.
 
-You can capture from everywhere, both within Emacs and from other
-applications like your web browser, PDF viewer, etc.
+You can capture from anywhere -- from an Emacs buffer or from other
```
Should be `---` not `--`.

tecosaur commented on features.org:
```diff
@@ -305,19 +299,44 @@ Tutorial on [[https://orgmode.org/worg/org-tutorials/org-custom-agenda-commands.
 *** More
 
 More on [[file:manual/Capture.html][capturing]] (manual)
+* Emacs Extensible
```
I kind of prefer "Extremely Extensible" or similar, I wonder if this may slightly confuse non-Emacsers looking at this page.

tgbugs replied:
I changed the headline, but kept the Emacs version in the text. Let's see what the general take on it is and we can adjust if needs be.

tecosaur replied:
I feel this is an approvement. Let's leave this be for now.

tecosaur reviewed (commented):
Hope you find this useful

tgbugs committed:
features.org add dry wiki link and justify paragraphs

Unjustified text looked scraggly and was harder to read/follow,
I implemented it on a per paragraph basis for maximum control.

tgbugs commented:
Ok. I think features is in a generally good state, but I would like to convert the remaining static svg images to be based on text as I did with simple-syntax.svg. Do you have the original text for the agenda, clocking, and links svgs so that I can create exact copies?

Not remotely pressing, but it might make sense to use the same front for the animations if possible. JetBrains mono is an open font, but I'm not sure how easy it is to ensure that it has been installed. I would suggest DejaVu Sans Mono since it is the default on a number of systems, but this is very much something for the future.

tgbugs committed:
simple-syntax.svg missing comma

tgbugs commented:
I've moved this out of draft since I think we are nearly ready, and can leave updating the svgs for another pr.

tecosaur commented on features.org:
```diff
@@ -5,6 +5,9 @@
 
 #+begin_export html
 <style>
+  .justify {
+    text-align: justify;
+  }
```
I'm a bit wary of justification because I know that on mobile it can easily produce 'rivers of whitespace'

tgbugs replied:
I'm missing something in my build which causes it not to match what you have running on your version of the site so I can't the styles and layout to match exactly, but the text more or less looks on mobile. If you are worried about it we could disable that style below a certain screen width.

tecosaur commented:
I feel this is almost ready to merge. Thanks again for all your work! It's greatly appreciated.

tecosaur commented:
Ok, I think I'll merge unless you have any objections - and then fix any minor issues with later commits.

BTW, I've now written a build script
```lisp
#!/usr/bin/env sh
":"; exec emacs --quick --script "$0" -- "$@" # -*- mode: emacs-lisp; lexical-binding: t; -*-

(require 'ox)
(require 'ox-html)
(load "~/.emacs.d/.local/straight/repos/emacs-htmlize/htmlize.el") ; system-dependant
(require 'htmlize)

(setq
 org-html-style-default ""
 org-html-scripts ""
 org-html-htmlize-output-type 'css
 org-html-doctype "html5"
 org-html-html5-fancy t
 debug-on-error t)

(let ((org-files (directory-files-recursively default-directory "\\.org$")))
  (dolist (org-file org-files)
    (message "[0;34m• %s[90m" (file-name-nondirectory org-file))
    (with-current-buffer (find-file-literally org-file)
      (condition-case err
          (org-html-export-to-html)
        (error (message "  [0;31m%s[90m"  (error-message-string err)))))))
(message "[0m")
(kill-emacs 0)
```

tgbugs commented:
I think we're good to go for this round.

tecosaur commented:
Yea, this is what I was concerned might happen:
![image](https://user-images.githubusercontent.com/20903656/95686163-d49cb200-0c2e-11eb-8083-79257997250d.png)

tgbugs commented:
I think it should be possible to disable the justify style based on screen sizes as described here? https://stackoverflow.com/questions/15061520/responsive-css-styles-on-mobile-devices-only

tecosaur commented:
Yea, that's on maximum screen size (the column has a max-width). The issue really is that web brousers use a cheap+bad justification algorighm, and so some lines just seem to end up badly :(

tgbugs commented:
Ugh. Subtle encouragement to only short words >_<.
